Core/Core/EmbeddedWebPage/ViewModel/EmbeddedWebPageViewModelLive.swift

```diff
@@ -35,15 +35,24 @@ public class EmbeddedWebPageViewModelLive: EmbeddedWebPageViewModel {
      This method assumes that feature flags for the given context are already in CoreData.
      */
     public static func isRedesignEnabled(in context: Context) -> Bool {
-        var featureFlagContext = context
-
         if context.contextType == .group {
+            var featureFlagContext = context
             let group = AppEnvironment.shared.subscribe(GetGroup(groupID: context.id))
             if let courseID = group.first?.courseID {
                 featureFlagContext = Context.course(courseID)
             }
+            return AppEnvironment.shared.subscribe(
+                GetEnvironmentFeatureFlags(context: .currentUser)
+            ).isFeatureEnabled(.react_discussions_post) ||
+            AppEnvironment.shared.subscribe(
+                GetEnabledFeatureFlags(context: featureFlagContext)
+            ).isFeatureFlagEnabled(.discussionRedesign)
```

Comment:
Let's add an explanatory comment here, it took a while for me to understand why we need both of these. Maybe renaming the `featureFlagContext` to `courseContext` would help. I don't know if the second part of the condition makes sense if the group is not in a course, if not, we could move it out of this OR into the if let block.